algorithms/src/polycommit/kzg10/mod.rs

```diff
@@ -401,16 +398,38 @@ impl<E: PairingEngine> KZG10<E> {
             None
         };
 
-        Ok(Proof { w: w.to_affine(), random_v })
+        Ok(KZGProof { w: w.to_affine(), random_v })
+    }
+
+    /// On input a polynomial `p` in Lagrange basis, and a point `point`,
+    /// outputs an evaluation proof for the same.
+    pub fn open_lagrange(
+        lagrange_basis: &LagrangeBasis<E>,
+        domain_elements: &[E::Fr],
+        evaluations: &[E::Fr],
+        point: E::Fr,
+        evaluation_at_point: E::Fr,
+    ) -> Result<KZGProof<E>, PCError> {
+        Self::check_degree_is_too_large(evaluations.len() - 1, lagrange_basis.size())?;
+        assert_eq!(evaluations.len().next_power_of_two(), lagrange_basis.size());
```

Comment:
is `evaluations.len().next_power_of_two()` guaranteed to not overflow? otherwise `checked_next_power_of_two` would be a better pick

Reply:
though I don't suppose it would be equal to `lagrange_basis.size()` under overflow conditions, which is grounds for a panic due to `assert_eq`; in any case, it would be more meaningful error